### PR TITLE
[SW-1106] using custom launch actions from synchros2

### PR DESCRIPTION
**Pull request by tcappellari-bdai**
## Change Overview

Modifying the launch files to use the custom launch actions from `synchros2` (formerly `bdai_ros2_wrappers`)

## Testing Done

driver and examples run as expected

Please create a checklist of tests you plan to do and check off the ones that have been completed successfully. Ensure that ROS 2 tests use `domain_coordinator` to prevent port conflicts. Further guidance for testing can be found on the [ros utilities wiki](https://github.com/bdaiinstitute/ros_utilities/wiki/Testing-guidelines).

### CODE REVIEW

File: spot_description/launch/description.launch.py
Comment (khughes-bdai) [inline spot_description/launch/description.launch.py:1]: I'm wondering if it's worth it to change this file (and others in spot_description), given that we are planning to separate out spot_description into its own repo soon and might not want this to depend on `synchros2`?

**Comment by khughes-bdai** [comment]
There are a bunch of boolean launch args related to image publishing defined here https://github.com/bdaiinstitute/spot_ros2/blob/main/spot_driver/spot_driver/launch/spot_launch_helpers.py & imported into the main launchfiles that could also benefit from this

File: spot_driver/launch/spot_driver.launch.py [hunk 1]
```diff
@@ -10,13 +10,15 @@
...
 from launch_ros.substitutions import FindPackageShare
+from synchros2.launch.actions import DeclareBooleanLaunchArgument, update_sigterm_sigkill_timeout
 
 from spot_driver.launch.spot_launch_helpers import IMAGE_PUBLISHER_ARGS, declare_image_publisher_args, spot_has_arm
 
 THIS_PACKAGE = "spot_driver"
 
 
 def launch_setup(context: LaunchContext, ld: LaunchDescription) -> None:
+    update_sigterm_sigkill_timeout(ld)
```
Comment (khughes-bdai) [inline spot_driver/launch/spot_driver.launch.py:21]: question -- how does this change how the driver shuts down on robot?

File: spot_driver/launch/spot_driver.launch.py [hunk 2]
```diff
@@ -163,10 +165,9 @@ def generate_launch_description() -> launch.LaunchDescription:
         )
     )
     launch_args.append(
-        DeclareLaunchArgument(
+        DeclareBooleanLaunchArgument(
             "launch_rviz",
             default_value="False",
```
Comment (khughes-bdai) [inline spot_driver/launch/spot_driver.launch.py:170]: nit: to match the style of the other launch args
```suggestion
            default_value=False,
```

**Review by jcarpinelli-bdai — APPROVED** [review]

File: spot_driver/launch/spot_driver.launch.py (hunk 1 above)
Comment (tcappellari-bdai) [reply spot_driver/launch/spot_driver.launch.py:21]: It just increases the allotted time for shutting down. I didn't notice any side effects while testing but maybe?

File: spot_description/launch/description.launch.py
Comment (tcappellari-bdai) [reply spot_description/launch/description.launch.py:1]: I don't have a problem with it depending on synchros2 but also dont have a problem with it not. I'll remove it though to reduce unnecessary dependencies

File: spot_driver/launch/spot_driver.launch.py (hunk 1 above)
Comment (khughes-bdai) [reply spot_driver/launch/spot_driver.launch.py:21]: ah cool I was worried that if it decreased the allotted time for shutting down that some stuff might not shut down cleanly (esp in the ros2 control case)

**Review by khughes-bdai — APPROVED** [review]
LGTM

**Pull request merged**